enhancements/proposals/1978-scylladbdatacenter-api/README.md

```diff
@@ -0,0 +1,741 @@
+- Instead, having a single field specifying container image of used components,
+  there are two fields, one for the repository and the second for version. 
+  This unnecessary split is error-prone.
+- Kubernetes API Convention suggests using arrays instead of maps. The status of racks doesn't follow that rule.
+- `cpuSet` setting is removed, as it only controls stricter validation of allocated resources. 
+  This is unnecessary when there is documentation about performance tuning containing these recommendations.
+- `hostNetworking` support is discontinued and setting it becomes deprecated and discouraged by not having a field to set it in API. 
+  To preserve backward compatibility controllers will support enabling it by setting special annotation 
+  on `v1alpha1.ScyllaDBDatacenter`. 
+  Controller migrating from `v1.ScyllaCluster` to `v1alpha1.ScyllaDBDatacenter` will set this annotation when required.
+
+### Helm chart
+
+Existing `scylla` helm chart is going to use existing `v1.ScyllaCluster`. It has several differences between what's in CRD
+and what's in the CRD, hence it cannot be simply mapped one-to-one.
+New helm chart called `scylladbdatacenter` will simply copy `spec` from `values` to a new CRD solving this issue making it 
+easily extensible, while keeping schema of CRD and helm schema in sync.
+
+### Scylla Manager integration
+
+Scylla Manager tasks do not fit into single datacenter, as Manager tasks work on the entire cluster. 
+These are removed from the spec and status of the new API until a better alternative for these is available.
+Existing controller reconciling these will continue using `v1.ScyllaCluster`, users wanting to use these, 
+won't be able to transition to new API until an extension is available.
+
+### Topology placement
+
+Currently, to provide a placement rules for rack, users have to provide a very lengthy `nodeAffinity`. 
+Even the simplest rules require providing 10'ish lines with 4'ish indentation levels. 
+This makes the entire resource huge and UX isn't great. 
+To improve it, new field `toplogyLabelSelector` added on datacenter and rack level, 
+provides a label selector which will be translated into placement nodeAffinity rules selecting nodes matching these labels. 
+Labels selectors from datacenter and rack level will be merged into final node affinity.
+
+To prevent misuse of this simplification keys of provided label selector will be validated against `topology.kubernetes.io/*` rule, 
+and usage of both `nodeAffinity` and `topologyLabelSelector` will be forbidden by the webhook.
+Users wanting to construct more complex placement rules will have to do it using `placement` spec.
+
+### Resource model
+
+Storage `capacity` is going to be merged into `resources` field under `scyllaDB`, but other settings like metadata
+and storage class name, will stay in place. 
+This makes it to be in sync on how PVC template is defined.
```

Comment:
Most of the changes done in this proposal are about simplifying and making UX better. Simple case == simple CRD representation. If user wants to have more advanced setup, then there're fields for these. StorageClass is not required because user may want to use the default one which is not something uncommon, while capacity is required, the same as cpu and memory.
We can't support full volumeClaimTemplate (name, access mode, selector, volumeName, volumeMode...) so I don't think we should go that way.

Reply:
> while capacity is required, the same as cpu and memory.

I don't think that's on us to decide, if regular Pods run without it we shouldn't add more obstacles, unless it breaks without it.

E.g. You can set defaults for these with LimitRanges - https://kubernetes.io/docs/concepts/policy/limit-range/